File: 3/pset3b.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class pset3b {

    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner sc = new Scanner(file);
        char [][] map = new char [323][31];
        for(int i=0; i < 323; i++){
            String currentstr = sc.nextLine();
            for(int j=0; j < 31; j++){
                char chartoinput = currentstr.charAt(j);
                map[i][j] = chartoinput;
            }
        }
        sc.close();
        System.out.println(countTrees(map, 1, 1) * countTrees(map, 1, 3) * countTrees(map, 1, 5) * countTrees(map, 1, 7) * countTrees(map, 2, 1));
    }
    
    public static long countTrees(char [][] map, int rowinc, int colinc){
        long count = 0;
        int col = 0 ;
        int row = 0;
        while(row < 323) {
            if (map[row][col] == '#') {
                count++;
            }
            row += rowinc;
            col += colinc;
            if (col >= 31)
                col -= 31;
            if (row > 323)
                break;
        }
        System.out.println(count);
        return count;
    }
    
}

File: 3/pset3b.py
def compute(map, row, col):

    rowlen = len(map)
    collen = len(map[0])

    count = 0
    i = 0
    j = 0

    while True:
        if map[i][j] == '#':
            count += 1
        i += row
        j += col
        if j >= collen:
            j -= collen
        if i >= rowlen:
            break


    return count

with open("pset3.txt", "r") as f:
    map = f.read().splitlines()

print(compute(map, 1, 1) * compute(map, 1, 3) * compute(map, 1, 5) * compute(map, 1, 7) * compute(map, 2, 1))